yt/frontends/swift/io.py

```diff
@@ -0,0 +1,168 @@
+"""
+Swift data-file handling function
+
+
+
+
+"""
+from __future__ import print_function
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2018, yt Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+from yt.utilities.on_demand_imports import _h5py as h5py
+import numpy as np
+
+from yt.frontends.sph.io import \
+    IOHandlerSPH
+
+class IOHandlerSwift(IOHandlerSPH):
+    _dataset_type = "swift"
+
+    def __init__(self, ds, *args, **kwargs):
+        super(IOHandlerSwift, self).__init__(ds, *args, **kwargs)
+
+    def _read_fluid_selection(self, chunks, selector, fields, size):
+        raise NotImplementedError
+
+    def _read_particle_coords(self, chunks, ptf):
+        # This will read chunks and yield the results.
+        chunks = list(chunks)
+        data_files = set([])
+        for chunk in chunks:
+            for obj in chunk.objs:
+                data_files.update(obj.data_files)
+        for data_file in sorted(data_files, key=lambda x: x.filename):
+            si, ei = data_file.start, data_file.end
+            f = h5py.File(data_file.filename, "r")
+            # This double-reads
+            for ptype, field_list in sorted(ptf.items()):
+                if data_file.total_particles[ptype] == 0:
+                    continue
+                x = f["/%s/Coordinates" % ptype][si:ei, 0].astype("float64")
+                y = f["/%s/Coordinates" % ptype][si:ei, 1].astype("float64")
+                z = f["/%s/Coordinates" % ptype][si:ei, 2].astype("float64")
+                if ptype == self.ds._sph_ptype:
+                    hsml = f[
+                        "/%s/SmoothingLength" % ptype][si:ei].astype("float64")
```

Comment:
Smoothing length's should not need to be doubles (`float32` should be fine) -- I don't think the conversion is necessary (unless the internals of `yt` require this).

Reply:
Also: why not just call the `_get_smoothing_length` function here? Am I missing something?

Reply:
We keep smoothing lengths are float64 for better compatibility with a bunch of cython functions. But yeah, we should call `_get_smoothing_length`